private/model/api/api.go

```diff
@@ -496,6 +501,26 @@ func (a *API) ServiceID() string {
 	return a.serviceID
 }
 
+func (a *API) ProtocolCanonicalPackageName() string {
+	switch a.Metadata.Protocol {
+	case "rest-xml":
+		return "aws_restxml"
+	case "rest-json":
+		return "aws_restjson"
+	case "ec2":
+		return "aws_ec2query"
+	case "query":
+		return "aws_query"
+	case "xml":
+		return "aws_xml"
```

Comment:
XML is not a protocol on its own.  We probably want to remove this case from the switch.